projects/gfbio-app/src/app/oidc/oidc.component.ts

```diff
@@ -0,0 +1,68 @@
+import {Component, OnInit, Inject, OnDestroy} from '@angular/core';
+import {UserService, Config, User} from "wave-core";
+import {AppConfig} from "../app-config.service";
+import {first, SubscriptionLike} from "rxjs";
+
+@Component({
+    selector: 'wave-gfbio-login',
+    templateUrl: 'oidc.component.html',
+    styleUrls: ['./oidc.component.scss'],
+})
+export class OidcComponent implements OnInit, OnDestroy {
+    private userSubscription : SubscriptionLike | undefined;
+    private loginSubscription : SubscriptionLike | undefined;
+    private logoutSubscription : SubscriptionLike | undefined;
+
+    user? : User;
+
+    constructor(
+        @Inject(Config) private readonly config: AppConfig,
+        private readonly userService: UserService,) {
+    }
+
+    ngOnInit(): void {
+        this.userSubscription = this.userService
+            .getSessionStream()
+            .pipe(first())
```

Comment:
If you use `first`, then this subscription terminates after the first value.
This means you don't need to save it and destroy it at a later point.

Reply:
Reasoning:
If a user opens the login component and clicks the login button, the backend (for whatever reason) might take a moment to respond. If the user closes the login component, and starts doing something else, whenever the backend returns the call, the user will still be redirected to the OIDC provider.

I would expect that to be weird behavior, because I closed the login component. There are some discussions about it online, but I defer to your preference.

Reply:
Okay, I see.